src/borg/cache.py

```diff
@@ -514,6 +514,15 @@ def __init__(
         self.txn_active = False
 
         self.path = cache_dir(self.repository, path)
+        if os.path.exists(self.path) and not self.check_cache_integrity():
+            logger.warning("Rebuilding cache as some cache-files have gone missing")
+            if not os.path.exists(os.path.join(self.path, "chunks")):
+                self.recreate_cache_config(delete_existing=True)  # recreating to remove integrity data stored in config
+                self.chunks = ChunkIndex()
+                self.chunks.write(os.path.join(self.path, "chunks"))
```

Comment:
this will create an empty (and thus incorrect) chunks index on disk, how is that helpful?

also creation of that file is done in a rather unsafe way.

Reply:
Well, I was assuming that creating an empty chunks cache should be enough for ```borg create``` to continue and re-populate it as it normally would. 
1. Should we be entirely rebuilding it here?
2. How else would be creating the file. Is there another piece of code that I can take a look at?

Reply:
borg create needs the chunks cache in a valid state (refcounts, sizes for all chunks in the repo) and updates refcounts on that basis.